stream.go

```diff
@@ -217,21 +217,28 @@ func (s *StreamService) Dial(ctx context.Context, params *DialStreamParams) (*Co
 
 	go func() {
 		defer conn.Close()
+		var prevErr error = nil
 		for {
 			var msg StreamMessage
 			if err := conn.conn.ReadJSON(&msg); err != nil {
 				if e, ok := err.(*websocket.CloseError); ok {
 					if isRevError, revError := IsRevError(e.Code); isRevError {
 						conn.err <- revError
-						return
+					} else {
+						conn.err <- e
 					}
+					// if we recieve any CloseError the connection is closed and needs to be reestablished before reading can continue
+					return
 				}
 
-				if websocket.IsUnexpectedCloseError(err, websocket.CloseGoingAway, websocket.CloseAbnormalClosure) {
```

Comment:
I think this is part of the issue. https://pkg.go.dev/github.com/gorilla/websocket#IsUnexpectedCloseError

we list CloseGoingAway and  CloseAbnormalClosure as expected and then keep trying to read. 

but if the connection is closed. we need reestablish it.